Assets/Exteel/ExteelScripts/Photon/BuildMech.cs

```diff
@@ -99,76 +128,249 @@ public class BuildMech : Photon.MonoBehaviour {
+					break;
+				}
+			case "BRF025": {
+					weaponScripts[i] = new BRF025();
+					weapons[i].transform.Rotate(0f, 0f, 8f * ((i % 2) == 0 ? -1 : 1));
+					weapons[i].transform.rotation = Quaternion.Euler(new Vector3(0,180,0));
+					weapons [i].transform.SetParent (hands [i % 2]);
+					bulletPrefabs [i] = Resources.Load ("BRF025B") as GameObject;
+					ShotSounds [i] = Resources.Load ("Sounds/Zeus_Fire") as AudioClip;
+					break;
+				}
+		/*	case "BCN029": {
+					weaponScripts[i] = new BRF025();
+					weapons[i].transform.Rotate(0f, 0f, 8f * ((i % 2) == 0 ? -1 : 1));
+					weapons[i].transform.rotation = Quaternion.Euler(new Vector3(0,180,0));
+					weapons [i].transform.SetParent (hands [i % 2]);
+					bulletPrefabs [i] = Resources.Load ("BRF025B") as GameObject;
+					ShotSounds [i] = Resources.Load ("Sounds/Planet_Fire") as AudioClip;
+					break;
+				}*/
+			case "SGN150": {
+					weaponScripts[i] = new SGN150();
+					weapons[i].transform.Rotate(0f, 0f, 8f * ((i % 2) == 0 ? -1 : 1));
+					weapons[i].transform.rotation = Quaternion.Euler(new Vector3(90,180,0));
+					weapons [i].transform.SetParent (hands [i % 2]);
+					bulletPrefabs [i] = Resources.Load ("SGN150B") as GameObject;
+					//ShotSounds [i] = Resources.Load ("Sounds/Planet_Fire") as AudioClip;
+					break;
+				}
+			case "RCL034":{
+					//Since the launch button is on right hand
+					p = new Vector3 (hands [1].position.x, hands [1].position.y - 0.4f, hands [1].position.z);
+					weapons [i].transform.SetParent (hands [1]);
+					weaponScripts[i] = new RCL034();
+
+					weapons[i].transform.rotation = Quaternion.Euler(new Vector3(-90,180,0));
+					weapons[i].transform.position = new Vector3(p.x , p.y , p.z);
+					bulletPrefabs [i] = Resources.Load ("RCL034B")  as GameObject;
+					ShotSounds [i] = Resources.Load ("Sounds/Hell_Fire") as AudioClip;
+
+					if(i==weaponOffset){
+						if(animator!=null){
+							//In Hangar or Lobby
+							animator.SetBool ("UsingRCL",true);
+						}else {
+							//In game
+							animator = GetComponentInChildren<MeleeCombat> ().GetComponent<Animator> ();
+							animator.SetBool ("UsingRCL",true);
+						}
+					}
+					weaponScripts [i + 1] = new EmptyWeapon ();
+					weapons [i + 1] = Instantiate (Resources.Load ("EmptyWeapon") as GameObject, p, transform.rotation) as GameObject;
+					weapons [i+1].transform.SetParent (hands [i % 2]);
+					weapons [i + 1].SetActive (false);
+					bulletPrefabs [i+1] = null;
+
+					i++;//skip the right hand
 					break;
 				}
 			}
-			weapons [i].transform.SetParent (hands [i % 2]);
 		}
-		weaponOffset = 0;
-		weapons [2].SetActive (false);
-		weapons [3].SetActive (false);
+		UpdateCurWeapons ();
+		weapons [(weaponOffset+2)%4].SetActive (false);
+		weapons [(weaponOffset+3)%4].SetActive (false);
 	}
 
+
 	public void EquipWeapon(string weapon, int weapPos) {
+		//if previous is two-handed => also destroy left hand 
```

Comment:
Make sure to test over network (so other players can see your weapons change, etc)